app/src/main/java/com/karumi/data/repository/SuperHeroRepository.kt

```diff
@@ -1,140 +1,18 @@
-                    + "is also a master of armed combat, able to use a variety of weapons but prefers "
-                    + "unarmed combat. He is a master planner who always thinks several steps ahead and "
-                    + "will go to extreme measures to achieve his goals and protect the kingdom "
-                    + "of Wakanda."),
-
-            SuperHero(name = "Captain America",
-                photo = "http://x.annihil.us/u/prod/marvel/i/mg/9/80/537ba5b368b7d.jpg",
-                isAvenger = true,
-                description = "Captain America represented the pinnacle of human physical perfection. He " +
-                    "experienced a time when he was augmented to superhuman levels, but generally" +
-                    " performed just below superhuman levels for most of his career. Captain America had a very high "
-                    + "intelligence as well as agility, strength, speed, endurance, and reaction time "
-                    + "superior to any Olympic athlete who ever competed."),
-
-            SuperHero(name = "Winter Soldier",
-                photo = "https://i.annihil.us/u/prod/marvel/i/mg/7/40/537bca868687c.jpg",
-                isAvenger = false,
-                description = "Olympic-class athlete and exceptional acrobat highly skilled in both unarmed" +
-                    " and armed "
-                    + "hand-to-hand combat and extremely accurate marksman. he is fluent in four languages "
-                    + "including German and Russian."),
-
-            SuperHero(name = "Captain Marvel",
-                photo = "https://x.annihil.us/u/prod/marvel/i/mg/6/30/537ba61b764b4.jpg",
-                isAvenger = false,
-                description = " Ms. Marvel's current powers include flight, enhanced strength, durability " +
-                    "and the ability to shoot concussive energy bursts from her hands."),
-
-            SuperHero(name = "Iron Fist",
-                photo = "https://i.annihil.us/u/prod/marvel/i/mg/6/60/537bb1756cd26.jpg",
-                isAvenger = false,
-                description = "Through concentration, Iron Fist can harness his spiritual energy, or chi, " +
-                    "to augment his physical and mental capabilities to peak human levels. By focusing his chi "
-                    + "into his hand, he can tap the superhuman energy of Shou-Lao and temporarily "
-                    + "render his fist superhumanly powerful, immune to pain and injury; however, this "
-                    + "process is mentally draining, and he usually needs recovery time before he can "
-                    + "repeat it. Iron Fist can heal himself of any injury or illness and project this "
-                    + "power to heal others.")
-
-        )
-    }
+    fun getByName(key: String): Either<DomainError, SuperHero> =
+        datasources.first { it.isUpdated() && it.contains(key) }.get(key)
```

Comment:
I don't like have the contains method, do you have other solution?

Reply:
As we discussed in person, it might worth a try to use `sequenceOf` or `streamOf` as I think they are lazy (meaning that if we use `get` it might only obtain the first result if it's present)